partner_firstname/models.py

```diff
@@ -155,7 +155,12 @@ def _get_inverse_name(self, name, is_company=False):
     def _inverse_name(self):
         """Try to revert the effect of :meth:`._compute_name`."""
         parts = self._get_inverse_name(self.name, self.is_company)
-        self.lastname, self.firstname = parts["lastname"], parts["firstname"]
+        values = {
+            "lastname": parts["lastname"],
+            "firstname": parts["firstname"]
+        }
+        self.write(values)
```

Comment:
@hbrunn , I would argue that not in all cases. Isin't update supposed to work with in-memory records? (solely). This of course works fine in the interface (because it updates the viewed model) which is later written on save with updated values.

But the problem is when you want to switch the firstname and lastname places. Imagine the following situation:

Clean database,  except for some partners with the following names:
1. Candyshop [is_company=True]
2. Bobby [is_company=False]
3. John Smith [is_company=False]

Now the user installs module partner_firstname into his db, the following computation takes place:
1. lastname: Candyshop, firstname: false, name: Candyshop
2. lastname: Bobby, firstname: false, name: Bobby
3. lastname: John, firstname: Smith, name: John Smith

Now the 2,3 are incorrect. I want to correct lastname and firstname for some records. One way to achieve this (and as suggested in the _get_inverse_name docstring) is to override that function and change the ordering there. To reuse the logic, I would get the result dictionary, {"lastname":..} and switch the places of the values. E.g. something like:

```python
@api.model
 def _get_inverse_name(self, name, is_company=False):
    res = super(ResPartner, self)._get_inverse_name(
        name, is_company=is_company)
    return {'firstname': res['lastname'], 'lastname': res['firstname']}
```
In the new module, I would also call the `_inverse_name` method to recompute the existing fields.

Now the problem in such a case would occur for entries 1 and 2. Because at the moment lastname write is made, the record does not have firstname yet (it was just returned from the function), and the constraints added will be violated (also failing to install the module).

Now I do realize, that there are ways to overcome this with post_install or migrations. Sadly, in some cases this requires some hand holding and the update is not totally automatic.I figured it would have been nicer to just write that place in one go and solve it all together. Unfortunately, as it later turned out this solution is incorrect.

I hope I made valid arguments that there is a problem in some cases.  Maybe anybody has experience how to do this differently. At the very least, an improvement on the IMO deceiving docstring would be nice?